modules/xfem/src/base/XFEM.C

```diff
@@ -882,6 +929,143 @@ XFEM::initCutIntersectionEdge(
   return does_intersect;
 }
 
+bool
+XFEM::healMesh()
+{
+  bool mesh_changed = false;
+
+  std::set<Node *> nodes_to_delete;
+  std::set<Node *> nodes_to_delete_displaced;
+  std::set<unsigned int> cutelems_to_delete;
+
+  for (unsigned int i = 0; i < _geometric_cuts.size(); ++i)
+  {
+    if (_geometric_cuts[i]->healMesh())
+    {
+      for (auto & it : _sibling_elems[_geometric_cuts[i]->getInterfaceID()])
+      {
+        Elem * elem1 = const_cast<Elem *>(it.first);
```

Comment:
I guess this scheme will just keep the stateful material properties for elem1 in the combined element.  Maybe we could average the properties from the two elements or something, although that has its problems.
I wonder if we could check to see if there are stateful properties when we heal and warn that they might not be right if healing is used. That could be done in a follow-on commit.

Reply:
That is a good point. I would like to address it in a follow-on PR.